odbc-api/src/connection.rs

```diff
@@ -293,6 +296,231 @@ impl<'c> Connection<'c> {
+            nullable: false,
+        };
+
+        let null_i32 = BufferDescription {
+            kind: BufferKind::I32,
+            nullable: true,
+        };
+        // The definitions for these descriptions are taken from the documentation of `SQLColumns`
+        // located at https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlcolumns-function
+        let catalog_name_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: self.max_catalog_name_len()?,
+            },
+            nullable: true,
+        };
+
+        let schema_name_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: self.max_schema_name_len()?,
+            },
+            nullable: true,
+        };
+
+        let table_name_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: self.max_table_name_len()?,
+            },
+            nullable: false,
+        };
+
+        let column_name_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: self.max_column_name_len()?,
+            },
+            nullable: false,
+        };
+
+        let data_type_desc = not_null_i16;
+
+        const MAX_EXPECTED_TYPE_NAME_LEN: usize = 255;
+        let type_name_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: MAX_EXPECTED_TYPE_NAME_LEN,
+            },
+            nullable: false,
+        };
+
+        let column_size_desc = null_i32;
+        let buffer_len_desc = null_i32;
+        let decimal_digits_desc = null_i16;
+        let precision_radix_desc = null_i16;
+        let nullable_desc = not_null_i16;
+
+        const MAX_EXPECTED_REMARKS_LEN: usize = 255;
+        let remarks_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: MAX_EXPECTED_REMARKS_LEN,
+            },
+            nullable: true,
+        };
+
+        const MAX_EXPECTED_COLUMN_DEFAULT_LEN: usize = 255;
+        let column_default_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: MAX_EXPECTED_COLUMN_DEFAULT_LEN,
+            },
+            nullable: true,
+        };
+
+        let sql_data_type_desc = not_null_i16;
+        let sql_datetime_sub_desc = null_i16;
+        let char_octet_len_desc = null_i32;
+        let ordinal_pos_desc = BufferDescription {
+            kind: BufferKind::I32,
+            nullable: false,
+        };
+
+        const MAX_EXPECTED_IS_NULLABLE_LEN: usize = 10;
+        let is_nullable_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: MAX_EXPECTED_IS_NULLABLE_LEN,
+            },
+            nullable: true,
+        };
+
+        let buffer_descs = [
+            catalog_name_desc,
+            schema_name_desc,
+            table_name_desc,
+            column_name_desc,
+            data_type_desc, // Currently unused
```

Comment:
If we don't want to include them in this PR, we could also skip binding these and use column indices instead